.circleci/config.yml

```diff
@@ -16,7 +16,502 @@ aliases:
     branches:
       only: master
 
+executors:
+  grafana-build:
+    docker:
+      - image: grafana/build-container:1.2.13
+  grafana-publish:
+    docker:
+      - image: grafana/grafana-ci-deploy:1.2.3
+  docker:
+    machine:
+      image: circleci/classic:201808-01
+
 jobs:
+  install-grabpl:
+    description: Install the Grafana Build Pipeline tool
+    executor: grafana-build
+    steps:
+      - run: |
+          echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | \
+            tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+          curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+          apt-get update && apt-get install -f google-cloud-sdk
+          echo $GCP_GRAFANA_UPLOAD_KEY > /tmp/gcpkey.json
+          gsutil cp gs://grafana-build-pipeline/grabpl bin/grabpl
+          chmod +x bin/grabpl
+      - persist_to_workspace:
+          root: .
+          paths:
+            - bin/grabpl
+
+  build-oss-backend:
+    description: "Build a certain variant of Grafana open-source back-end binaries"
+    parameters:
+      variant:
+        type: string
+    executor: grafana-build
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: Build Grafana backend
+          # TODO: Don't use $CIRCLE_TAG for master
+          command: /tmp/workspace/bin/grabpl build-backend --variants << parameters.variant >> $CIRCLE_TAG
+      - run:
+          name: Move artifacts
+          command: mkdir -p oss && mv bin oss/
+      - persist_to_workspace:
+          root: .
+          paths:
+            - oss/bin/*
+      - run:
+          name: CI job failed
+          command: ./scripts/ci-job-failed.sh
+          when: on_fail
+      - run:
+          name: CI job succeeded
+          command: ./scripts/ci-job-succeeded.sh
+          when: on_success
+
+  build-oss-frontend:
+    executor: grafana-build
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: Build Grafana frontend
+          command: /tmp/workspace/bin/grabpl build-frontend
+      - run:
+          name: Move artifacts
```

Comment:
I'm wondering if it might be a good idea to incorporate the knowledge of what artifacts are produced in a build step so that grabpl can copy them to the correct location instead of that having to be a separate step in the ci script. Could have an `--ouput oss` arg or something like that.

Reply:
@xlson That's probably right. This was hacked together in order to please `persist_to_workspace`, since it doesn't give you particularly good control of where to place files in the workspace, and it's also badly documented.